### ISSUE TITLE
Use GetWithContext in apm pkg

### ISSUE THREAD

**Issue by Tarliton** [newrelic/newrelic-client-go#721]
### Feature Description
Hello, 
Could we get a version of `apm` pkg functions that use `context.Context`?

For example, I have a instrumented `http.RoundTripper` that I have configured with `ConfigHTTPTransport`, but when I call functions from `apm` pkg I am not seeing those requests in New Relic Transaction.

I believe the reason is that we are not calling it with the context for the request.

If that is not the reason, please could you guys point me in the right direction?

### Describe Alternatives
Maybe something like [this](https://github.com/newrelic/newrelic-client-go/blob/2932f5d6275d9017fd1ce5764d2de258575dd187/pkg/nerdgraph/nerdgraph.go#L42-L50).

### Additional context
I would like to see in New Relic Transactions what endpoints I am calling when I use this library.

Thank you

**Comment by zlesnr** [comment]
I believe this could be done in a pretty straight forward manner.  Similar to how we do in the `alerts` package, as well is elsewhere probably.

https://github.com/newrelic/newrelic-client-go/blob/main/pkg/alerts/channels.go#L77-L102

Where the non-context-aware method calls the context-aware method to keep compatibility with older implementations, while at the same time providing new context methods to allow what you are after and keeping the logic the same.

Based on what I see here, it doesn't look like too many methods would need to be modified for this:
https://pkg.go.dev/github.com/newrelic/newrelic-client-go@v0.58.5/pkg/apm

Is this a contribution you'd like to make @Tarliton?

And to confirm, are you seeing transactions for the other packages that do implement the context handling?

**Comment by Tarliton** [comment]
Hello @zlesnr, 

Thank you for your reply. Yes, I would like to contribute.

I will also check if we can see transactions when we use methods with context handling (I do not use any of them now, hence I can't answer at the moment).


To confirm, I will create a new `*WithContext` function for each APM struct function. In this case it will be a new function for all the following functions:
```go
func (a *APM) CreateDeployment(applicationID int, deployment Deployment) (*Deployment, error)
func (a *APM) CreateLabel(label Label) (*Label, error)
func (a *APM) DeleteApplication(applicationID int) (*Application, error)
func (a *APM) DeleteDeployment(applicationID int, deploymentID int) (*Deployment, error)
func (a *APM) DeleteLabel(key string) (*Label, error)
func (a *APM) GetApplication(applicationID int) (*Application, error)
func (a *APM) GetApplicationInstance(applicationID int, instanceID int) (*ApplicationInstance, error)
func (a *APM) GetKeyTransaction(id int) (*KeyTransaction, error)
func (a *APM) GetLabel(key string) (*Label, error)
func (a *APM) GetMetricData(applicationID int, params MetricDataParams) ([]*MetricData, error)
func (a *APM) GetMetricNames(applicationID int, params MetricNamesParams) ([]*MetricName, error)
func (a *APM) ListApplicationInstances(applicationID int, params *ListApplicationInstancesParams) ([]*ApplicationInstance, error)
func (a *APM) ListApplications(params *ListApplicationsParams) ([]*Application, error)
func (a *APM) ListDeployments(applicationID int) ([]*Deployment, error)
func (a *APM) ListKeyTransactions(params *ListKeyTransactionsParams) ([]*KeyTransaction, error)
func (a *APM) ListLabels() ([]*Label, error)
func (a *APM) UpdateApplication(applicationID int, params UpdateApplicationParams) (*Application, error)
```

Is that correct?

Thanks

**Comment by zlesnr** [comment]
That sounds right to me @Tarliton.  That way we can keep the current interface and behavior, as well as provide context methods future facing using the same behavior.  Looking forward to your contribution, thank you.

**Comment by Tarliton** [comment]
Hello @zlesnr,

I have tested my PR using the new `WithContext` function.
After calling the function I was not seeing the expected external call to api.newrelic.com.

I was able to find out why.

As I said, I am passing a instrumented `http.RoundTripper` to `ConfigHTTPTransport`. That works okay, but when NewRelic client create a httpClient [here](https://github.com/newrelic/newrelic-client-go/blob/main/internal/http/client.go#L63-L69) it checks if that http.RoundTripper is an specific implementation, in this case the implementation from std http lib (*http.Transport). If we change that code to accept any implementation that implements the RoudTripper interface, it works.

If I change that part to:
```go
...
if cfg.HTTPTransport != nil {
	c.Transport = cfg.HTTPTransport
}
...
```

I am able to see the calls to api.newrelic.com in New Relic Transactions.

Is that the right fix? Should I make a PR for that?

Thanks

**Comment by zlesnr** [comment]
I took a brief look and I'm unclear why we try to match a specific implementation.  I see that there are tests in place for the `ConfigHTTPTransport` also, so perhaps there is an update needed to test the situation you are mentioning.  I don't see a reason why we wouldn't want to take the snippet above, so long as we could include a test that covers that case.

**Comment by Tarliton** [comment]
Hello, I could make a new PR changing that.

But, do you have any suggestion about how would we test that?
I mean, we can continue with the tests that we already have, but I couldn't come up with a new test to add...

**Comment by zlesnr** [comment]
It looks like we do have some usage in our tests currently.  See here for example:

https://github.com/newrelic/newrelic-client-go/blob/main/newrelic/newrelic_test.go#L123
https://github.com/newrelic/newrelic-client-go/blob/main/newrelic/newrelic_test.go#L132

If you run into issues getting the tests to pass after making the change, we can see what the coverage looks like at that point and add something more specific to the case.

**Comment by zlesnr** [comment]
This looks related also: https://github.com/newrelic/newrelic-client-go/pull/688

**Comment by ctrombley** [comment]
Merged #727 and fixed the transport issue in #744.  Thanks again @Tarliton !
